core/src/main/java/org/ehcache/spi/ServiceLocator.java

```diff
@@ -145,8 +136,17 @@ void addService(final Service service, final boolean expectsAbstractRegistration
         }
       }
 
+      if (service.getClass().isAnnotationPresent(ServiceDependency.class)) {
```

Comment:
Yep, especially given the implementation of `isAnnotationPresent`. Still, I prefer the current style - feels more "correct" and less `null` tied.